lib/animina_web/live/root_live.ex

```diff
@@ -71,21 +92,39 @@ defmodule AniminaWeb.RootLive do
         ]}
       />
 
-      <.form :let={f} for={@form} action={@action} method="POST" class="space-y-6">
+      <.form
+        :let={f}
+        id="basic_user_form"
+        for={@form}
+        action={@action}
+        phx-trigger-action={@trigger_action}
+        method="POST"
+        class="space-y-6"
+        phx-change="validate"
+        phx-submit="submit"
+      >
         <div>
           <label for="username" class="block text-sm font-medium leading-6 text-gray-900">
             <%= gettext("Username") %>
           </label>
           <div class="mt-2">
-            <%= text_input(f, :username,
+            <%= text_input(
+              f,
+              :username,
               class:
-                "block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6",
+                "block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset  placeholder:text-gray-400 focus:ring-2 focus:ring-inset  sm:text-sm sm:leading-6 " <>
+                  if(get_field_errors(f[:username], :username) != [],
```

Comment:
I haven't played with this but I think `if(get_field_errors(f[:username], :username) != []` can be done easier with something like `unless(get_field_errors(f[:username], :username)`

```elixir
iex(1)> if [] do
...(1)> IO.puts "a"
...(1)> else
...(1)> IO.puts "b"
...(1)> end
a
:ok
``

Reply:
I couldn't get unless to work like you suggested `unless(get_field_errors(f[:username], :username))` , instead I've opted to include `== []`  in the check.  `unless(get_field_errors(f[:username], :username) == [])`